compiler/scenario-service/client/src/DA/Daml/LF/ScenarioServiceClient/LowLevel.hs

```diff
@@ -121,9 +121,13 @@ instance NFData Error where
     rnf = rwhnf
 
 findServerJar :: IO FilePath
-findServerJar = do
-  runfilesDir <- locateRunfiles (mainWorkspace </> "compiler/scenario-service/server")
-  pure (runfilesDir </> "scenario-service.jar")
+findServerJar = locateResource Resource
+  { runfilesPath = mainWorkspace </> "compiler" </> "scenario-service" </> "server" </> "scenario-service.jar"
+  , resourcesPath = "scenario-service.jar"
```

Comment:
It seems that in every use case of `locateResource`, the `resourcesPath` equals the simple filename suffix of the `runFilesPath`.

Would it make sense to capture this invariant (if it is) by having `Resource` contain a `runfilesPathPrefix` member instead of `runfilesPath`. And then appending the `resourcesPath` simple filename to determine the `runfilesPath`?

Reply:
I also noticed this pattern, but at first I avoided pursuing it since sometimes the wanted resource is a directory, so the resources path isn't just the last filename component of the runfiles path; e.g. to get [`pkg-db_dir`](https://github.com/digital-asset/daml/pull/15446/commits/44a8b76ccc1e15cff287a5b566ad3b0b2ebff8ae), this doesn't work

```haskell
locateResource Resource
  { runfilesPath = mainWorkspace </> "compiler" </> "damlc" </> "pkg-db" </> "pkg-db_dir"
  , resourcesPath = ""
  }
```

because in a packaged app it will just return the top level `"resources"` directory. In answering this, however, I noticed that it is still true that the `resourcesPath` is a suffix of the `runfilesPath`, even if not always just the filename component. Flipping the API to be resource-first it makes more sense: "give me the path for this in the `"resources"` directory, and also give me the prefix for that in case this is a `bazel run` target and I need to fetch it from the runfiles."

Thanks for making me realize this, PR incoming :)

Reply:
:+1: